MonoDevelop.FSharpBinding/FakeSearchCategory.fs

```diff
@@ -0,0 +1,160 @@
+                                              "FAKE task runner", 
+                                              "Center Bottom", 
+                                              IconId("md-command"))
+            if pad <> null then Some(pad)
+            else None
+    with exn -> None
+
+  member x.Run (baseDirectory, task, buildScript) =
+    let fsiProcess = 
+      let startInfo = 
+        new ProcessStartInfo
+          (FileName = buildScript, UseShellExecute = false, Arguments = task, 
+           RedirectStandardError = true, CreateNoWindow = true, RedirectStandardOutput = true,
+           RedirectStandardInput = true, StandardErrorEncoding = Text.Encoding.UTF8, 
+           StandardOutputEncoding = Text.Encoding.UTF8,
+           WorkingDirectory = baseDirectory)
+      view.WriteOutput(sprintf "FAKE task runner: Starting %s %s" buildScript task)
+      view.Clear()
+
+      try
+        Process.Start(startInfo)
+      with e ->
+        LoggingService.LogDebug (sprintf "FAKE task runner %s" (e.ToString()))
+        reraise()
+    do 
+      Event.merge fsiProcess.OutputDataReceived fsiProcess.ErrorDataReceived
+        |> Event.filter (fun de -> de.Data <> null)
+        |> Event.add (fun de -> 
+            LoggingService.LogDebug (sprintf "Interactive: received %s" de.Data)
+
+            async {
+              do! DispatchService.GuiDispatch(fun _ -> view.WriteOutput (de.Data + "\n"))
+                  |> Async.AwaitTask
+            } |> Async.RunSynchronously)
+
+      fsiProcess.EnableRaisingEvents <- true
+      fsiProcess.BeginOutputReadLine()  
+      fsiProcess.BeginErrorReadLine()
+
+  override x.Control : Gtk.Widget = view :> Gtk.Widget
+  override x.Initialize(_container:MonoDevelop.Ide.Gui.IPadWindow) =
+    x.UpdateColors()
+    x.UpdateFont()
+
+  member x.UpdateColors() =
+    match view.Child with
+    | :? Gtk.TextView as v -> 
+          let colourStyles = Mono.TextEditor.Highlighting.SyntaxModeService.GetColorStyle(MonoDevelop.Ide.IdeApp.Preferences.ColorScheme.Value)
+          
+          let shouldMatch = PropertyService.Get ("FSharpBinding.MatchWithThemePropName", false)
+          let themeTextColour = colourStyles.PlainText.Foreground |> cairoToGdk
+          let themeBackColour = colourStyles.PlainText.Background |> cairoToGdk
+
+          if shouldMatch then
+              v.ModifyText(Gtk.StateType.Normal, themeTextColour)
+              v.ModifyBase(Gtk.StateType.Normal, themeBackColour)
+          else
+              let textColour = PropertyService.Get ("FSharpBinding.TextColorPropName", "#000000") |> ColorHelpers.strToColor
+              let backColour = PropertyService.Get ("FSharpBinding.BaseColorPropName", "#FFFFFF") |> ColorHelpers.strToColor
+              v.ModifyText(Gtk.StateType.Normal, textColour)
+              v.ModifyBase(Gtk.StateType.Normal, backColour)
+    | _ -> ()
+    
+  member x.UpdateFont() = 
+    let fontName = MonoDevelop.Ide.Fonts.FontService.MonospaceFont.Family
```

Comment:
Do we need a custom font, why not the font in Preferences | Environment | Fonts ?